src/main/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension.java

```diff
@@ -149,4 +241,22 @@ private static Store getStore(ExtensionContext extensionContext) {
     return extensionContext.getStore(SOFT_ASSERTIONS_EXTENSION_NAMESPACE);
   }
 
+  public static AssertionErrorCollector getAssertionErrorCollector(ExtensionContext context) {
+    return getStore(context).getOrComputeIfAbsent(ConcreteSoftAssertions.class, unused -> new ConcreteSoftAssertions(),
+                                                  ConcreteSoftAssertions.class);
+  }
+
+  private static <T extends SoftAssertionsProvider> T instantiateProvider(ExtensionContext context,
+                                                                          Class<T> concreteSoftAssertionsProviderType) {
+    T provider = ReflectionSupport.newInstance(concreteSoftAssertionsProviderType);
+    provider.setDelegate(getAssertionErrorCollector(context));
+    return provider;
+  }
+
+  public static <T extends SoftAssertionsProvider> T getSoftAssertionsProvider(ExtensionContext context,
```

Comment:
In a word: no. :smile:

The purpose of making this public is to form an API so that third-party extensions that want to leverage this extension can do so without having to peek inside the internals of `ExtensionContext`. See here: https://github.com/osgi/osgi-test/blob/e2e99949cfbf61b225209440ab8265e67a7745df/org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceExtension.java#L149-L159

...for an example of this usage pattern in action - the `ServiceExtension` uses the `BundleContextExtension`. This way the user doesn't have to be careful about the order in which they declare/initialise the two extensions - they both use the same API function to instantiate the field that they want to inject. We've found that it works quite well in osgi-test.

By way of full disclosure: I'm hoping to do a similar thing to chain the `SoftAssertionsExtension` with an extension that implements Mockito soft assertions, so that the other extension can get a handle on the current AssertionErrorCollector for the current test context and log Mockito verification failures into it.

Reply:
Done.